source/base/quadrature_lib.cc

```diff
@@ -1014,6 +1014,116 @@ QWeddle<dim>::QWeddle ()
   Quadrature<dim> (QWeddle<dim-1>(), QWeddle<1>())
 {}
 
+  template <int dim>
+  QTelles<dim>::QTelles (
+      const Quadrature<1> base_quad, const Point<dim> singularity)
+      :
+  /**
+  * We need the explicit implementation if dim == 1. If dim > 1 we use the
+  * former implementation and apply a tensorial product to obtain the higher
+  * dimensions.
+  **/
+      Quadrature<dim>(
+          dim == 2 ?
+          QAnisotropic<dim>(
+              QTelles<1>(base_quad, Point<1>(singularity[0])),
+              QTelles<1>(base_quad, Point<1>(singularity[1]))) :
+          dim == 3 ?
+          QAnisotropic<dim>(
+              QTelles<1>(base_quad, Point<1>(singularity[0])),
+              QTelles<1>(base_quad, Point<1>(singularity[1])),
+              QTelles<1>(base_quad, Point<1>(singularity[2]))) :
+          Quadrature<dim>())
+  {
+  }
+
+  template <int dim>
+  QTelles<dim>::QTelles (
+      const unsigned int n, const Point<dim> singularity)
+      :
+  /**
+  * In this case we map the standard Gauss Legendre formula using the given
+  * singularity point coordinates.
+  **/
+      Quadrature<dim>(QTelles<dim>(QGauss<1>(n), singularity))
+  {}
+
+
+
+  template <>
+  QTelles<1>::QTelles (
+      const Quadrature<1> base_quad, const Point<1> singularity)
+      :
+  /**
+  * We explicitly implement the Telles' variable change if dim == 1.
+  **/
+      Quadrature<1>(base_quad)
+  {
+      /**
+      * We define all the constants to be used in the implementation of
+      * Telles' rule
+      **/
+      double eta_bar = singularity[0] * 2 - 1;
+      double eta_star = eta_bar * eta_bar - 1;
+      double gamma_bar;
+
+      std::vector<Point<1>> quadrature_points_dummy(quadrature_points.size());
+      std::vector<double> weights_dummy(weights.size());
+      unsigned int cont=0;
+      double toll=1e-5;
+      for(unsigned int d=0; d<quadrature_points.size(); ++d)
+      {
+          if(std::abs(quadrature_points[d][0]-singularity[0])>toll)
+          {
+              quadrature_points_dummy[d-cont]=quadrature_points[d];
+              weights_dummy[d-cont]=weights[d];
+          }
+          else
+          {
+              cont=1;
+          }
+
+      }
+      if(cont==1)
+      {
+          quadrature_points.resize(quadrature_points_dummy.size()-1);
+          weights.resize(weights_dummy.size()-1);
+          for(unsigned int d=0; d<quadrature_points.size()-1; ++d)
+          {
+              quadrature_points[d]=quadrature_points_dummy[d];
+              weights[d]=weights_dummy[d];
+          }
+      }
+
+      if((eta_bar * eta_star + std::abs(eta_star))==0 || (eta_bar * eta_star - std::abs(eta_star))==0)
```

Comment:
What is this check for? A floating point comparison with ==0 doesn't make a lot of sense (will rarely happen). You probably need a check with an epsilon here.